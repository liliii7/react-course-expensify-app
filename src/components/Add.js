import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";
class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    console.log(expense);
    this.props.addExpense(expense);
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} 
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  
  addExpense: (expense) => {
    console.log("expense",expense);
    console.log('dispatch',dispatch)
    dispatch(addExpense(expense))
  }
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
