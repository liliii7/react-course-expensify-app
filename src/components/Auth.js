import React from "react";
import { connect } from "react-redux";
import { startSetExpenses } from "../actions/expenses";
import { firebase } from "../firebase/firebase";
import { withRouter } from "../WithRouter"; 

class AuthHandler extends React.Component {

  componentDidMount() {
    this.unsubscribe = firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        this.props.startSetExpenses().then(() => {
          if (window.location.pathname === '/') {
            this.props.navigate('/dashboard'); 
          }
          console.log('log in');
        });
      } else {
        this.props.navigate('/'); 
        console.log('log out');
      }
    });
  }

  componentWillUnmount() {
    if (this.unsubscribe) {
      this.unsubscribe();
    }
  }

  render() {
    return this.props.children;
  }
}

const mapDispatchToProps = (dispatch) => ({
  startSetExpenses: () => dispatch(startSetExpenses())
});

export default withRouter(connect(null, mapDispatchToProps)(AuthHandler));
