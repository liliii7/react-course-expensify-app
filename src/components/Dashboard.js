import React from "react";
import ExpenseList from "./ExpensesList";
import ExpenseListFitlers from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";
export const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFitlers />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
