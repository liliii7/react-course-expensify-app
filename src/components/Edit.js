import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, startRemoveExpense } from "../actions/expenses";
import { useParams, useNavigate } from "react-router-dom";

const EditExpensePage = (props) => {
  const { id } = useParams();
  const navigate = useNavigate();

  const onSubmit = (expense) => {
    props.editExpense(id, expense);
    navigate('/');
  };

  const onRemove = () => {
    props.startRemoveExpense({ id });
    navigate('/');
  };

  const expense = props.expenses.find((expense) => expense.id === id);

  return (
    <div>
      <ExpenseForm onSubmit={onSubmit} expense={expense} />
      <button onClick={onRemove}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state) => ({
  expenses: state.expenses
});

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
