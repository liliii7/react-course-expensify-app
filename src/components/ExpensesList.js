import React, { useEffect } from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpensesListItem";
import selectExpensese from "../selectors/expenses";

const ExpenseList = (props) => {

  useEffect(() =>{
    console.log(props)
  }, [props]);

  return (
  <div>
    {props.expenses.length < 0 ? (
      <p>No expenses</p>
    ) : (
      props.expenses.map((expense) => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
  </div>
)};  

const MapStateToProps = (state) => {
  return {
    expenses: selectExpensese(state.expenses, state.filters),
  };
};

export default connect(MapStateToProps)(ExpenseList);
