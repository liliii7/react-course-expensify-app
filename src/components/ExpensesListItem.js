import React from "react";
import { Link } from "react-router-dom";

export const ExpenseListItem = ({description, amount, createdAt, id}) => (
    <div>
    <Link to={`/edit/${id}`}>
    <h1>{description}</h1>
    </Link>
    <p>{amount} - {createdAt}</p>
    </div>
);


export default ExpenseListItem;