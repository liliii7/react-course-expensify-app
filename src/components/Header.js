import React from "react";
import { NavLink } from "react-router-dom";


const Header = () => (
  <header>
    <h3>Expensify</h3>
    <NavLink to="/"  activeclassname="is-active" exact='true'>
      home
    </NavLink>
    <NavLink to="/create" activeclassname="is-active">
      create
    </NavLink>
    <NavLink to="/edit" activeclassname="is-active">
      edit
    </NavLink>
    <NavLink to="/help" activeclassname="is-active">
      help
    </NavLink>
  </header>
);

export default Header;
