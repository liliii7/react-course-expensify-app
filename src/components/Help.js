import React from "react";

const HelpPage = () => <div>help expense component</div>;

export default HelpPage;