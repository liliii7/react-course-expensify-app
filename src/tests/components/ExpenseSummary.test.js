import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test('should correctly render Expenses Summary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={125} />)
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render Expenses Summary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={500} />)
    expect(wrapper).toMatchSnapshot();
})