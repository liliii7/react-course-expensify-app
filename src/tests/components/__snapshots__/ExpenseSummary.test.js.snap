// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expenses Summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.25
  </h1>
</div>
`;

exports[`should correctly render Expenses Summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $5.00
  </h1>
</div>
`;
